Add Mistral Large 3 to nightly CI tests

alisonshao opened the pull request:
## Summary
- Add nightly performance test for `mistralai/Mistral-Large-3-675B-Instruct-2512`
- Test configuration:
  - TP=8
  - `--attention-backend trtllm_mla`
  - `--model-loader-extra-config '{"enable_multithread_load": true}'`
  - `--chat-template mistral`
  - `SGLANG_ENABLE_JIT_DEEPGEMM=0`

## Test plan
- [ ] Verify nightly CI workflow runs successfully on 8-gpu-h200 runner
- [ ] Check performance traces are published correctly

alisonshao committed:
Add Mistral Large 3 to nightly CI tests

Add nightly performance test for mistralai/Mistral-Large-3-675B-Instruct-2512 model with:
- TP=8
- trtllm_mla attention backend
- Multithread loading enabled
- Mistral chat template
- SGLANG_ENABLE_JIT_DEEPGEMM=0

alisonshao commented:
testing at: https://github.com/sgl-project/sglang/actions/runs/19948050892/job/57201949750
waiting for result

alisonshao committed:
Add CI register for nightly test

alisonshao committed:
Fix: Move Mistral-Large-3 test to B200 runner (trtllm_mla requires Blackwell)

alisonshao committed:
Move Mistral-Large-3 test to first position in 8-gpu-b200 workflow

- Move Mistral test to run immediately after dependency installation
- Add `if: always()` to DeepSeek v3.1 since it's no longer first
- Remove duplicate Mistral test from end of workflow

This allows faster debugging of the Mistral test and ensures other
tests continue running regardless of Mistral test outcome.

alisonshao committed:
Fix race condition in weight validation cleanup

Multiple TP ranks calling find_local_hf_snapshot_dir simultaneously
would all detect corrupted/missing files and race to delete the same
cache directory. One rank would succeed, others would fail with
"No such file or directory" errors.

Fix by wrapping the validation and cleanup logic in a file lock
(using suffix "-validation" to avoid conflict with download lock).
After acquiring the lock, re-check if the snapshot dir still exists
since another process may have already cleaned it up.

alisonshao committed:
Add pre-download step for Mistral-Large-3 model

Download the model before running the test to avoid server launch
timeout issues. The 120-minute timeout allows for full download on
first run. Subsequent runs will use the cached model and skip quickly.

alisonshao committed:
Reduce pre-download timeout to 60 minutes

ispobock commented on test/nightly/test_mistral_large3_perf.py:
```diff
@@ -0,0 +1,67 @@
+        if "SGLANG_ENABLE_JIT_DEEPGEMM" in os.environ:
+            del os.environ["SGLANG_ENABLE_JIT_DEEPGEMM"]
+
+    def test_bench_one_batch(self):
```
can we also add accuracy test for it?

alisonshao replied:
just added test_accuracy_mgsm method using the same eval framework as other text models. Set initial threshold to 0.90 (placeholder). Will calibrate the threshold after observing actual model performance.

alisonshao commented on python/sglang/srt/model_loader/weight_utils.py:
```diff
@@ -309,8 +311,22 @@ def find_local_hf_snapshot_dir(
+    # validating and cleaning up the same model cache simultaneously.
+    # This prevents race conditions where multiple ranks detect corruption
+    # and try to delete the same files at the same time.
+    with get_lock(model_name_or_path, cache_dir, suffix="-validation"):
```
it's a general bug fix discovered while debugging the Mistral test. When model weights are corrupted/missing, all 8 TP ranks detect the issue simultaneously and race to delete the cache directory. One rank succeeds, others fail with "No such file or directory" errors. This was causing CI failures.
the fix adds a file lock so only one rank performs validation and cleanup at a time. other models just happened to have complete caches so the bug wasn't triggered.

alisonshao commented on .github/workflows/nightly-test-nvidia.yml:
```diff
@@ -420,7 +420,35 @@ jobs:
         run: |
           IS_BLACKWELL=1 bash scripts/ci/ci_install_dependency.sh
 
+      - name: Pre-download Mistral-Large-3 model
```
We can remove it now, if preferred.
Other models (DeepSeek, Qwen3-235B, etc.) were already fully cached on the B200 runner from previous runs. Mistral is a newly added model that wasn't cached yet, so I added this step to support the first time running this model.

alisonshao committed:
Add MGSM accuracy test for Mistral Large 3

Added test_accuracy_mgsm method using the same eval framework as other
text models. Set initial threshold to 0.90 (placeholder). Will calibrate
the threshold after observing actual model performance.

alisonshao committed:
Remove pre-download step (model is now cached)
